misc/package/Data Files/MWSE/core/mcm/components/categories/Category.lua

```diff
@@ -100,10 +104,18 @@ end
 --- @param components mwseMCMComponent.getComponent.componentData[]
 function Category:createSubcomponents(parentBlock, components)
 	for _, component in pairs(components or {}) do
-		component.parentComponent = self
-		local newComponent = self:getComponent(component)
 
-		newComponent:create(parentBlock)
+		-- Make sure it's actually a `Component`.
+		if not component.componentType then
+			local componentClass = fileUtils.getComponentClass(component.class)
+			if not componentClass then
+				error(string.format("Could not intialize component %q", component.label))
```

Comment:
This is not the same as `mwse.log` that was used in the `getComponent`. The `error` function stops the execution of the current script, while `mwse.log` will only log an error message, with the script execution continuing.

Reply:
I should have mentioned this in the PR, but I'm not sure of a nice way to change this to a `mwse.log` statement without using the `goto` operator, and I've heard `goto` should be avoided.

The problem is that is `component` is invalid and we can't find the right `componentClass`, then it might not make sense to call `component:create(parentBlock)`. But this is happening outside the `if not component.componentType` check.

Maybe I could set `component = nil` if the check fails, and then add a `nil` check before `component:create`? e.g.  
```lua
-- Make sure it's actually a `Component`.
if not component.componentType then
	local componentClass = fileUtils.getComponentClass(component.class)
	if not componentClass then
		mwse.log("Could not intialize component %q", component.label)
		component = nil
	end
	component.parentComponent = self
	componentClass:new(component) -- Modifies in-place, which is why it's okay to use in this loop.
end
if component then
	--- @cast component mwseMCMComponent
	component:create(parentBlock)
end
```
But this also seems weird. I thought it might be best to just stop the execution of the MCM if such a state is reached, but I'm not sure if this is the best way forward.

It's worth mentioning that the previous implementation would error out, so adding an explicit `error` statement wouldn't change the behavior of `Category:createSubcomponents`. It would just make the error a bit more explicit.

This is because the previous implementation would result in the `mwse.log` message getting printed from inside `getComponent`, but then `getCompoent` would return `nil`, resulting in 
```lua
-- inside Category:createSubcomponents:
local newComponent = self:getComponent(component) -- `nil` is returned

newComponent:create(parentBlock) -- ERROR: trying to call `nil:create(parentBlock)`
```

Reply:
Oh, I see you inspected the workings of the old API more in-depth than I did. Your reasoning checks out.

Reply:
Sounds good! I'll keep the `error` statemetns in then.